onadata/apps/logger/tests/models/test_instance.py

```diff
@@ -263,3 +264,14 @@ def test_instance_json_updated_on_review(self):
         self.assertEqual(SubmissionReview.APPROVED, status)
         self.assertEqual("Hey There", comment)
         self.assertTrue(instance.has_a_review)
+
+    def test_numeric_checker_with_negative_integer_values(self):
+        # Evaluate negative integer values
+        string_value = "-16"
+        result = numeric_checker(string_value)
+        self.assertEqual(result, -16)
+
+        # Evaluate un-signed integer values
+        string_value = "16"
+        result = numeric_checker(string_value)
+        self.assertEqual(result, 16)
```

Comment:
Can we also add checks for a `float` value, `nan` value and a non-numeric string for the `numeric_checker` function? It does not seem like we test them anywhere else.